### PR TITLE
Added implementation of values

### PR DESCRIPTION
**Pull request by cperezln**
# Feature

- Please insert link to associated issue here: [/issues/15](https://github.com/hablapps/pykx/issues/15)

## What does this change introduce?
An implementation of the `values` function: https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.values.html

This is intended to be a 1:1 implementation of the `values` function from pandas.

## General

- [ ] Has an example been added to demo the new feature?
- [ ] Have existing examples been updated or tested?
- [ ] Have you added any new Environment Variables/Configuration Options? If yes please tick the boxes below as applicable
    - [ ] Addition to reimporter logic within `src/pykx/pykx.q` and `src/pykx/reimporter.py`
    - [ ] Have updated the `src/pykx/util.py` logic which is used for environment variable
- [ ] If there have been any dependency updates have they been reflected in all files?
    - [ ] pyproject.toml
    - [ ] docs/getting-started/installing.md
    - [ ] conda-recipe/meta.yaml
    - [ ] README.md
- [ ] If any examples have been updated has it's associated `.zip` been updated

## Code

- [x] Has all temporary code used during development been removed?
- [x] Has all commented out (unused) code been removed?
- [x] Where reasonable have you ensured there is no duplication of existing code?
- [x] If applicable for your use-case have you ensured that the code is performant?

## Testing

- [x] Have unit tests been created or existing ones updated to test this new functionality?

## Documentation

- [x] Has documentation been added for all public code?
- [ ] Has a release note been included for the new feature?
- [ ] Has any documentation which would benefit from this feature been updated to use the most up to date functionality?
- [ ] If a new class has been added has a documentation stub `.md` file associated with it been created?
- [ ] If any documentation page has been created has it been added to `mkdocs.yml`
- [x] Have you checked your changes with a spell checker? (US English)

### CODE REVIEW

**Review by chraberturas — APPROVED** [review]

File: src/pykx/pandas_api/pandas_meta.py [hunk 1]
```diff
@@ -134,6 +134,11 @@ def shape(self):
     @property
     def size(self):
         return q('{count[x] * count[cols x]}', self)
+    
+    @property
+    def values(self):
+        tab = self
+        return q('value peach', tab)
```
Comment (nipsn) [inline src/pykx/pandas_api/pandas_meta.py:141]: Not completely sure `peach` does anything here. I would have to ask the devs if multithreading inside the q session is supported in PyKX, since I have seen little use of `peach` on the library and by default q runs singlethreaded (see https://code.kx.com/q/wp/multi-thread/#setting-up-kdb-for-parallel-processing). If it turns out it's not really used, then (minor) consider either of these implementations:

```python
kx.q("(value')", q_table)
```
Or its pythonic counterpart:

```python
kx.q.each(kx.q.value, q_table)
```
If it turns out that multithreading inside the q session is supported, then (minor) consider the pythonic implementation:

```python
kx.q.peach(kx.q.value, q_table)
```

Either way, I will try to ask this question to the devs next time we talk to them and answer this comment with whatever they tell me so you can decide what to do next.

File: docs/user-guide/advanced/Pandas_API.ipynb [hunk 2]
```diff
@@ -353,6 +353,32 @@
...
+    "### Table.values\n",
+    "Return a matricial representation of the DataFrame.\n",
+    "\n",
+    "Only the values in the DataFrame will be returned, the axes labels will be removed.\n",
+    "\n",
+    "**Returns:**\n",
+    "\n",
+    "| Type              | Description                                                          |\n",
+    "| :---------------: | :------------------------------------------------------------------- |\n",
+    "| List | Rows of the table, each as a list. |"
```
Comment (nipsn) [inline docs/user-guide/advanced/Pandas_API.ipynb:370]: For the description I would instead write something like "The values of the table as a list of list" or "The values of the table as a matrix" or something among those lines. The current description looks a bit confusing to me personally.

File: tests/test_pandas_api.py [hunk 3]
```diff
@@ -2029,3 +2029,12 @@ def test_keyed_loc_fixes(q):
         mkt[['k1', 'y']]
     with pytest.raises(KeyError):
         mkt['k1']
+
+def test_values(q):
```
Comment (nipsn) [inline tests/test_pandas_api.py:2033]: Running `pflake8` returns "E302 expected 2 blank lines, found 1" on this line.

File: src/pykx/pandas_api/pandas_meta.py [hunk 4]
```diff
@@ -134,6 +134,11 @@ def shape(self):
     @property
     def size(self):
         return q('{count[x] * count[cols x]}', self)
+
```
Comment (nipsn) [inline src/pykx/pandas_api/pandas_meta.py:137]: Running `pflake8` returns "W293 blank line contains whitespace" on this line.

**Review by marcosvm13 — APPROVED** [review]

**Review by nipsn — CHANGES_REQUESTED** [review]
Overall the implementation is on the right track, but there are a couple of aspects that need to be checked. Mainly:

- Does the q session run with multithreading enabled on PyKX? I will ask the devs and get back to you so you can decide what to do next.
- The tests need a bit of work. I think a specific case with null values would be helpful to see if they are handled properly. 
- The assertion itself it's a bit tricky, so feel free to iterate over my suggestions.

File: docs/user-guide/advanced/Pandas_API.ipynb [hunk 5]
```diff
@@ -353,6 +353,32 @@
...
+   "cell_type": "markdown",
+   "id": "4023811f-0f23-4e4b-919c-c055b865d9d2",
+   "metadata": {},
+   "source": [
+    "### Table.values\n",
+    "Return a matricial representation of the DataFrame.\n",
+    "\n",
+    "Only the values in the DataFrame will be returned, the axes labels will be removed.\n",
+    "\n",
+    "**Returns:**\n",
```
Comment (tortolavivo23) [inline docs/user-guide/advanced/Pandas_API.ipynb:366]: The "return" section in other functions that are of the `property` type is not present, as it can be in `columns` or `dtype`, so consider whether it is necessary to include it or if it adds nothing.

Comment (cperezln) [reply docs/user-guide/advanced/Pandas_API.ipynb:366]: I have tried to follow the actual Pandas documentation. That is the reason for the 'Returns' section. As can be seen here (https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.size.html), the 'size' method does not have a 'Returns' section, but the 'values' method does (https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.values.html).

File: docs/user-guide/advanced/Pandas_API.ipynb (hunk 2 above)
Comment (cperezln) [reply docs/user-guide/advanced/Pandas_API.ipynb:370]: I'd liked more the statement "The values of the table as a matrix", so I will use that one.

File: src/pykx/pandas_api/pandas_meta.py (hunk 4 above)
Comment (cperezln) [reply src/pykx/pandas_api/pandas_meta.py:137]: Working on it, thanks

File: tests/test_pandas_api.py [hunk 6]
```diff
@@ -2029,3 +2029,12 @@ def test_keyed_loc_fixes(q):
         mkt[['k1', 'y']]
     with pytest.raises(KeyError):
         mkt['k1']
+
+def test_values(q):
+    col1 = q('10?100')
+    col2 = q('10?`a`b`c`d')
+    col3 = q('10?`x`y`z`1`2`3')
+    tab = q('{[price; sym; id] ([]p: price; s: sym; i: string id)}', col1, col2, col3)
```
[the remark this answers is not in the document]
Comment (cperezln) [reply tests/test_pandas_api.py:2037]: Already figured it out, thanks!

File: tests/test_pandas_api.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (cperezln) [reply tests/test_pandas_api.py:2033]: Already working on it. Moved tests definition to line 60.

File: tests/test_pandas_api.py [hunk 7]
```diff
@@ -2029,3 +2029,12 @@ def test_keyed_loc_fixes(q):
...
         mkt['k1']
+
+def test_values(q):
+    col1 = q('10?100')
+    col2 = q('10?`a`b`c`d')
+    col3 = q('10?`x`y`z`1`2`3')
+    tab = q('{[price; sym; id] ([]p: price; s: sym; i: string id)}', col1, col2, col3)
+    q_table = tab
+    pandas_table = tab.pd()
+    assert pandas_table.values.tolist() == q_table.values.py()
```
[the remark this answers is not in the document]
Comment (cperezln) [reply tests/test_pandas_api.py:2040]: As you mentioned, the conversion is not perfect. When you execute tab.values.np(), it will not return a np.array but instead a np.array of np.arrays. I believe that, instead of altering the definition of the conversion to numpy or iterating through a list, it is a better option to keep it as it is.

File: src/pykx/pandas_api/pandas_meta.py (hunk 1 above)
Comment (nipsn) [reply src/pykx/pandas_api/pandas_meta.py]: Just asked the PyKX team about this. Turns out that the library itself handles the multithreading. Generally speaking PyKX is greedy, so it takes everything it can in terms of resources, but when dealing with q sessions we should focus on using `each`. If I understood correctly, in the future they will have a sort of flag that will change the behavior of `each` into `peach`.

TLDR: We should use `each`.

Feel free to use the solution you feel  fits best.

**Review by nipsn — APPROVED** [review]
Looks good to me now :+1:

File: docs/user-guide/advanced/Pandas_API.ipynb [hunk 8]
```diff
@@ -353,6 +353,32 @@
...
+    "\n",
+    "| Type              | Description                                                          |\n",
+    "| :---------------: | :------------------------------------------------------------------- |\n",
+    "| List | The values of the table as a matrix. |"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
```
Comment (tortolavivo23) [inline docs/user-guide/advanced/Pandas_API.ipynb:375]: The value of `execution_count` should be null, like in the others code cells

**Review by tortolavivo23 — CHANGES_REQUESTED** [review]
